### ISSUE TITLE
npm/yarn audit parser fails on CWE field

### ISSUE THREAD

**Issue by valentijnscholten** [DefectDojo/django-DefectDojo#5999]
I don't know if this is something new, but `yarn` imports are failing when multiple CWE values are references for an advisory:
```
          "cwe":"[\"CWE-918\",\"CWE-1333\"]",
```

**Comment by dvdgsng** [comment]
Experiencing the same with:

```
docker run -it node:16.13.0 yarn --version
1.22.15

```

**Comment by grohmio** [comment]
Same problem with single CWE values, e.g.

`"cwe":"[\"CWE-79\"]"`

**Comment by valentijnscholten** [comment]
this PR fixes that as well, covered by a unit test. I think yarn gets their advisories from upstream, which is where something might have changed. it's a weird format anyway.

**Comment by valentijnscholten** [comment]
Fixed in 'dev' / 2.9.0

**Comment by dvdgsng** [comment]
Thanks for fixing this so quickly, looking forward to v2.9.0 <3

**Comment by valentijnscholten** [comment]
Reopened as of today it seems `npm audit` reports are now suffering from the same issue: 
`"cwe": "[\"CWE-22\",\"CWE-59\"]",`

**Comment by valentijnscholten** [comment]
finally fixed also yarn.... now they have seemed to change the format again....

```
  File "./dojo/tools/utils.py", line 10, in get_npm_cwe
    if cwe_node.startswith('CWE-'):
AttributeError: 'list' object has no attribute 'startswith'
```

```
        "cwe": [
          "CWE-346",
          "CWE-453"
        ],
```

let's wait if this stabilizes.

**Comment by mtesauro** [comment]
Yeah, what a mess.

**Comment by valentijnscholten** [comment]
merged into `dev` for `2.9.0`
